functional/test/test_streams.py

```diff
@@ -145,3 +147,40 @@ def test_to_csv(self):
         sequence.to_csv(tmp_path)
         result = seq.csv(tmp_path).to_list()
         self.assertEqual(expect, result)
+
+    def test_to_sqlite3_failure(self):
+        # test failure case
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+        with self.assertRaises(ValueError):
+            seq(elements).to_sqlite3(1, insert_sql)
+
+    def test_to_sqlite3_file(self):
+        tmp_path = 'functional/test/data/tmp/{}.db'.format(time.time())
```

Comment:
I am going to guess that the `time.time()` is because the file wasn't getting removed between tests and causing problem? If thats the case, then using a random number would be better here. Could you clarify why `time.time()` is needed?